samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -106,11 +106,11 @@ public ApplicationStatus status(StreamApplication app) {
       }
 
       if (unsuccessfulFinish) {
-        return ApplicationStatus.UnsuccessfulFinish;
+        return ApplicationStatus.unsuccessfulFinish(null); // no exception for remote execution.
```

Comment:
Good suggestion! IN this case, I created a SamzaException with the diagonostic info. Please take a look. Thanks.